x-pack/plugins/cases/server/services/user_actions/index.ts

```diff
@@ -661,4 +671,48 @@ export class CaseUserActionService {
       },
     };
   }
+
+  public async getCaseUserActionStats({ caseId, filter }: { caseId: string; filter?: KueryNode }) {
+    const response = await this.context.unsecuredSavedObjectsClient.find<
+      CaseUserActionAttributesWithoutConnectorId,
+      UserActionsStatsAggsResult
+    >({
+      type: CASE_USER_ACTION_SAVED_OBJECT,
+      hasReference: { type: CASE_SAVED_OBJECT, id: caseId },
+      page: 1,
+      perPage: 1,
+      sortField: defaultSortField,
+      aggs: CaseUserActionService.buildUserActionStatsAgg(),
+      filter,
+    });
+
+    const result = {
+      total: response.total,
+      total_comments: 0,
+      total_other_actions: 0,
+    };
+
+    response.aggregations?.totals.buckets.forEach(({ key, doc_count: docCount }) => {
+      if (key === 'comment') {
+        result.total_comments = docCount;
+      }
+    });
+
+    result.total_other_actions = result.total - result.total_comments;
+
+    return result;
+  }
+
+  private static buildUserActionStatsAgg(): Record<
+    string,
+    estypes.AggregationsAggregationContainer
+  > {
+    return {
+      totals: {
```

Comment:
I think we need a little more filtering here. The doc (I'll message you it just in case) says that the `comment` filter will show only text related comments (the user typing a comment, a lens visualization, a timeline etc). The other types of "comments" like alerts should be counted and shown under the `actions` filter. So I think if we filter only on `attributes.type === comment` we'd accidentally include the stats for alerts and a few other types.

Instead we need to only count "user" comments for the comments filter and everything else for the actions filter. To get a count of only the "user" comments we need a filter like: `${CASE_USER_ACTION_SAVED_OBJECT}.attributes.payload.comment.type === ${CommentType.user} && ${CASE_USER_ACTION_SAVED_OBJECT}.attributes.type === comment`

To do that I think we need a [filter aggregation](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-filter-aggregation.html) something like:

```
{
      totals: {
        filter: {
          bool: {
            filter: [
              {
                term: {
                  [`${CASE_USER_ACTION_SAVED_OBJECT}.attributes.type`]: {
                    value: ActionTypes.comment,
                  },
                },
              },
              {
                term: {
                  [`${CASE_USER_ACTION_SAVED_OBJECT}.attributes.payload.comment.type`]: {
                    value: CommentType.user,
                  },
                },
              },
            ],
          },
        },
      },
    };
```

To use that format you'll likely need to update the `filter` aggregation schema in the core library here: https://github.com/elastic/kibana/blob/main/packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/aggregations/aggs_types/bucket_aggs.ts#L179

It'll likely need to look something like this (needs to look more pretty though):

```
  filter: s.oneOf([
    termSchema,
    existsSchema,
    s.object({
      bool: s.object({
        filter: s.arrayOf(termValueSchema),
      }),
    }),
  ]) as unknown as ObjectType,
```

I haven't tested any of that though 😆

Reply:
@jonathan-buttner What if we do

```
return {
      user_actions: {
          terms: {
              field: `${CASE_USER_ACTION_SAVED_OBJECT}.attributes.type`,
          },
      },
      comments: {
          terms: {
              field: `${CASE_USER_ACTION_SAVED_OBJECT}.attributes.payload.comment.type`,
          },
      },
```

to avoid filtering and have the stats in case we need them in the future?

Reply:
@cnasikas I am playing around with your suggestion and it fixes a bug I was having but I wanted to confirm something.

The actual comments should have `${CASE_USER_ACTION_SAVED_OBJECT}.attributes.payload.comment.type == 'user'` ?

Reply:
Yup, that was what I was thinking, thanks 👍 

And I noticed the whole saved object is returned with the call so this is probably a good place to add `fields: ['owner']` [like discussed here](https://github.com/elastic/kibana/pull/149269#discussion_r1089313686) right @jonathan-buttner ?

Reply:
Yes, you can do that if you like but is not necessary. Jonathan's scenario is a bit different because he wants to iterate through all documents and get the ids. In your case, the call returns only one document (we cannot avoid it I think) which we do not access it.

Reply:
> to avoid filtering and adding the new schema and have more stats in case we need them in the future?

@cnasikas ah good idea! @adcoelho we'll want to include a size parameter for each `terms` aggregation to ensure that we get more than the default 10 buckets: https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-terms-aggregation.html#search-aggregations-bucket-terms-aggregation-size

Maybe hard code it to like 100 for now or something.

@cnasikas what's the intention behind the `user_actions` aggregation? Is that to get the total? I think using the total on the find response works too right?

> And I noticed the whole saved object is returned with the call so this is probably a good place to add fields: ['owner'] https://github.com/elastic/kibana/pull/149269#discussion_r1089313686 right @jonathan-buttner ?

Yeah I agree with Christos, you can do it but it's not necessary. Ideally we'd pass a page size of 0 but I don't think core supports that like elasticsearch does.

Reply:
> @cnasikas what's the intention behind the user_actions aggregation? Is that to get the total? I think using the total on the find response works too right?

I noticed that too, I didn't include it in the solution and all tests still pass.

Reply:
> @cnasikas what's the intention behind the user_actions aggregation? Is that to get the total? I think using the total on the find response works too right?

Yes, it was for the total. You are right. The find response should be enough.